AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyAssetEditor.py

```diff
@@ -0,0 +1,163 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Object to house all the Qt Objects and behavior used in testing the asset editor
+"""
+
+from editor_python_test_tools.utils import TestHelper as helper
+from PySide2 import QtWidgets, QtCore, QtTest
+import pyside_utils
+import azlmbr.editor as editor
+import azlmbr.bus as bus
+import os
+from editor_python_test_tools.QtPyCommon import QtPyCommon
+from consts.asset_editor import (ASSET_EDITOR_UI, EVENTS_QT, DEFAULT_SCRIPT_EVENT, DEFAULT_METHOD_NAME)
+from consts.general import (WAIT_TIME_SEC_3, SAVE_STRING)
+
+
+class QtPyAssetEditor(QtPyCommon):
+    """
+    QtPy class for manipulating the Asset editor UI and performing actions such as creating Script Events and
+    saving them to disk.
+    """
+
+    def __init__(self, editor_main_window: QtWidgets.QMainWindow):
+        super().__init__()
+        self.asset_editor = editor_main_window.findChild(QtWidgets.QDockWidget, ASSET_EDITOR_UI)
+        self.asset_editor_widget = self.asset_editor.findChild(QtWidgets.QWidget, "AssetEditorWindowClass")
+        self.asset_editor_row_container = self.asset_editor_widget.findChild(QtWidgets.QWidget, "ContainerForRows")
+        self.menu_bar = self.asset_editor_widget.findChild(QtWidgets.QMenuBar)
+
+    def add_method_to_script_event(self, method_name: str) -> None:
+        """
+        function for adding a method to a script event. This mimics clicking the + button on the right side of the UI
+        then entering text into the Name field.
+
+        params method_name: The name you want to give the new method
+
+        returns none
+        """
+        add_event = self.asset_editor_row_container.findChild(QtWidgets.QFrame, EVENTS_QT).findChild(
+            QtWidgets.QToolButton, "")
+
+        assert add_event is not None, "Failed to add new method to script event."
+
+        add_event.click()
+
+        helper.wait_for_condition(lambda: self.asset_editor_widget.findChild(
+            QtWidgets.QFrame, DEFAULT_SCRIPT_EVENT) is not None, WAIT_TIME_SEC_3)
+
+        # Categories need to be expanded before we can manipulate fields hidden within
+        self.expand_category_by_name(DEFAULT_SCRIPT_EVENT)
+        self.expand_category_by_name("Name")
+        self.update_new_method_name(DEFAULT_METHOD_NAME, method_name)
+
+    def update_new_method_name(self, old_method_name: str, updated_method_name: str) -> None:
+        """
+        Function for finding a default script event and changing its name
+
+        param: method_name: the name you want to give the new method.
+
+        Returns: None
+        """
+        children = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, "Name")
+        method_name_field = ""
+        old_method_exists = False
+        for child in children:
```

Comment:
White space here to make the loop easier to read over at a glance due to the nested control structure (if-statement).

Reply:
broke up this logic a bit so the loop and the control structure have a little extra space and are easier to read over.